nemo/tests/test.rs

```diff
@@ -0,0 +1,446 @@
+)]
+fn weakly_sticky_negative(fixture: Fixture<&str>) {
+    test(fixture, false, &RulesProperties::is_weakly_sticky)
+}
+
+#[dir_test(
+    dir: "$CARGO_MANIFEST_DIR/../resources/testcases_static_checks/tests/positive/isWeaklySticky",
+    glob: "*.rls",
+    postfix: "weakly_sticky_positive",
+)]
+fn weakly_sticky_positive(fixture: Fixture<&str>) {
+    test(fixture, true, &RulesProperties::is_weakly_sticky)
+}
+
+struct TestCase<'a> {
+    expected_result: bool,
+    rule_set: RuleSet,
+    static_check: &'a dyn Fn(&RuleSet) -> bool,
+}
+
+impl<'a> TestCase<'a> {
+    fn test_from_rule_file(
+        rule_file: PathBuf,
+        expected_result: bool,
+        static_check: &'a dyn Fn(&RuleSet) -> bool,
+    ) -> Result<Self, Box<dyn std::error::Error>> {
+        Ok(Self {
+            expected_result,
+            rule_set: run_until_rule_model(rule_file)?,
+            static_check,
+        })
+    }
+
+    fn run(&self) -> Result<(), Box<dyn std::error::Error>> {
+        let func = self.static_check;
+        assert_eq!(self.expected_result, func(&self.rule_set));
+        Ok(())
+    }
+}
+
+fn run_until_rule_model(rule_file: PathBuf) -> Result<RuleSet, Box<dyn std::error::Error>> {
+    // TimedCode::instance().start();
+    // TimedCode::instance().sub("Reading & Preprocessing").start();
+
+    log::info!("Parsing rules ...");
+
+    // if cli.rules.len() > 1 {
+    //     return Err(CliError::MultipleFilesNotImplemented);
+    // }
+
+    // let program_file = cli.rules.pop().ok_or(CliError::NoInput)?;
+    let program_filename = rule_file.to_string_lossy().to_string();
+    let program_content = read_to_string(rule_file.clone())?;
+    //     .map_err(|err| CliError::IoReading {
+    //     error: err,
+    //     filename: program_filename.clone(),
+    // })?;
+
+    let program_ast = match nemo::parser::Parser::initialize(
+        &program_content,
+        program_filename.clone(),
+    )
+    .parse()
+    {
+        Ok(program) => program,
+        Err((_program, report)) => {
+            report.eprint()?;
+            return Err(Box::new(Error::new(ErrorKind::Other, "error1")));
+        }
+    };
+
+    let program = match nemo::rule_model::translation::ASTProgramTranslation::initialize(
+        &program_content,
+        program_filename.clone(),
+    )
+    .translate(&program_ast)
+    {
+        Ok(program) => program,
+        Err(report) => {
+            report.eprint()?;
+            return Err(Box::new(Error::new(ErrorKind::Other, "error2")));
+            // return Err(CliError::ProgramParsing {
+            //     filename: program_filename,
+            // });
+        }
+    };
+
+    // override_exports(&mut program, cli.output.export_setting);
+    // log::info!("Rules parsed");
+    let rule_set: RuleSet = program.rules().cloned().collect();
+    Ok(rule_set)
+}
+
+fn path_canonicalized(path: &str) -> PathBuf {
+    PathBuf::from_str(path).unwrap().canonicalize().unwrap()
```

Comment:
In general: use `expect(message)` instead of `unwrap` where `message` explains why this operation cannot fail under normal circumstances.